### DISCUSSION TITLE
login is required to use the 'clickhouse' driver

### DISCUSSION THREAD

**Discussion by ClaudioTuio** [ariga/atlas#3095] [Q&A] [answered]
# Replicating ClickHouse Schema Application

## Introduction

Good morning! I am trying to replicate the guide provided by [Atlas](https://atlasgo.io/guides/clickhouse?utm_source=clickhouse&utm_term=knowledge).

## Issue

I am connected but still not able to apply the schema.

## Steps to Reproduce

1. Login to Atlas:
    ```sh
    ➜ atlas login
    ```
    Output:
    ```
    You are now connected to migration-houseclick on Atlas Cloud.
    ```

2. Apply the schema:
    ```sh
    ➜ sudo atlas schema apply \
        -u "clickhouse://localhost:9000/demo" \
        --to file://schema.hcl \
        --dev-url "docker://clickhouse/23.11/demo"
    ```
    Error:
    ```
    Error: timeout: login is required to use the 'clickhouse' driver: <nil>
    ```

## Conclusion

The error indicates a timeout and suggests that login is required to use the 'clickhouse' driver. Further investigation is needed to resolve this issue.

**Comment by a8m** [comment, marked as answer]
Hey! Since you also posted it on our Discord, let's keep discussion in a single place please.
Ref: https://discord.com/channels/930720389120794674/1224649302761013288
